### PR TITLE
Use ansi escape code to indicate download progress in taskbar

### PR DESCRIPTION
**Pull request by xPaw**
20240728-BDJ1Lf18

Ref: https://learn.microsoft.com/en-us/windows/terminal/tutorials/progress-bar-sequences (this is based on conenum, a few terminals support it)

### CODE REVIEW

File: DepotDownloader/Ansi.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    public enum ProgressState
+    {
+        Hidden = 0,
+        Default = 1,
+        Error = 2,
+        Indeterminate = 3,
+        Warning = 4,
+    }
+
+    const string ESC = "\u001b";
```
Comment (yaakov-h) [inline DepotDownloader/Ansi.cs:19]: I think this is `\e` in modern C#

File: DepotDownloader/AnsiDetector.cs [hunk 2]
```diff
@@ -0,0 +1,133 @@
+// Copied from https://github.com/spectreconsole/spectre.console/blob/d79e6adc5f8e637fb35c88f987023ffda6707243/src/Spectre.Console/Internal/Backends/Ansi/AnsiDetector.cs
```
Comment (yaakov-h) [inline DepotDownloader/AnsiDetector.cs:1]: Licensed under MIT

File: DepotDownloader/AnsiDetector.cs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+// which is based on https://github.com/keqingrong/supports-ansi/blob/master/index.js
+// <auto-generated/>
+
+using System;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+
+namespace Spectre.Console;
```
Comment (yaakov-h) [inline DepotDownloader/AnsiDetector.cs:10]: Should we just use/ref their NuGet package?

Comment (xPaw) [reply DepotDownloader/AnsiDetector.cs:10]: I wanted to use their package for displaying a fancier progress, but the downloader code got refactored that its not very realisting to get per file progress now, don't think it's worth pulling the whole pkg for it.

File: DepotDownloader/Ansi.cs (hunk 1 above)
Comment (xPaw) [reply DepotDownloader/Ansi.cs]: it complains about string escaping being a preview feature or something, i changed these consts to chars.

**Pull request merged**